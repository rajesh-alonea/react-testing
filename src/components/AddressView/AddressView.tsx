import React from 'react';
import { Address } from '../../models';

export interface AddressViewProps {
  address: Address;
}

export const AddressView = ({ address }: AddressViewProps) => {
  const {
    firstName,
    lastName,
    company,
    address: street,
    city,
    state,
    zip,
  } = address;

  return (
    <div>
      <h3 className="m-0">
        {firstName} {lastName}
      </h3>
      {company !== undefined ? <p className="m-0">{company}</p> : null}
      <p className="m-0">{street}</p>
      <p className="m-0">
        {city}, {state} {zip}
      </p>
    </div>
  );
};
