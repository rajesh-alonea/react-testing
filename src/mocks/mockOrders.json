[
  {
    "id": "3f9db14f-9d21-455a-b250-dbe28b05202a",
    "createdAt": "2021-01-01T14:00:00.000Z",
    "items": [
      {
        "productId": "apple-imac",
        "productName": "iMac",
        "price": 1299,
        "quantity": 1
      },
      {
        "productId": "apple-ipad",
        "productName": "iPad",
        "price": 249,
        "quantity": 1
      },
      {
        "productId": "nest-learning-thermostat",
        "productName": "Nest Learning Thermostat",
        "price": 249,
        "quantity": 1
      }
    ],
    "shippingAddress": {
      "firstName": "Elon",
      "lastName": "Musk",
      "company": "Tesla, Inc.",
      "address": "3500 Deer Creek Road",
      "city": "Palo Alto",
      "state": "CA",
      "zip": "94304"
    }
  },
  {
    "id": "68e19edd-8a5e-4af2-a432-2e7ab04e7dcd",
    "createdAt": "2021-07-09T14:00:00.000Z",
    "items": [
      {
        "productId": "apple-imac",
        "productName": "iMac",
        "price": 1299,
        "quantity": 1
      },
      {
        "productId": "apple-ipad",
        "productName": "iPad",
        "price": 249,
        "quantity": 1
      },
      {
        "productId": "nest-learning-thermostat",
        "productName": "Nest Learning Thermostat",
        "price": 249,
        "quantity": 1
      },
      {
        "productId": "gopro-hero",
        "productName": "GoPro HERO Session Action Camera",
        "price": 149,
        "quantity": 4
      }
    ],
    "shippingAddress": {
      "firstName": "Tim",
      "lastName": "Cook",
      "company": "Apple, Inc.",
      "address": "One Apple Park Way",
      "city": "Cupertino",
      "state": "CA",
      "zip": "95014"
    }
  },
  {
    "id": "70871df2-7b8a-40fc-bb33-6d0d9af73dff",
    "createdAt": "2021-03-05T14:00:00.000Z",
    "items": [
      {
        "productId": "canon-eos-rebel",
        "productName": "Canon EOS Rebel SL1",
        "price": 403,
        "quantity": 1
      }
    ],
    "shippingAddress": {
      "firstName": "Mark",
      "lastName": "Zuckerberg",
      "company": "Facebook",
      "address": "1 Hacker Way",
      "city": "Menlo Park",
      "state": "CA",
      "zip": "94025"
    }
  },
  {
    "id": "7df4f6d1-d5a0-454c-9bed-a6284e84c372",
    "createdAt": "2021-02-05T14:00:00.000Z",
    "items": [
      {
        "productId": "fitbit-charge-hr",
        "productName": "Fitbit Charge HR",
        "price": 136,
        "quantity": 1
      }
    ],
    "shippingAddress": {
      "firstName": "Jeff",
      "lastName": "Bezos",
      "company": "Amazon.com, Inc.",
      "address": "2111 7th Ave",
      "city": "Seattle",
      "state": "WA",
      "zip": "98121"
    }
  }
]
