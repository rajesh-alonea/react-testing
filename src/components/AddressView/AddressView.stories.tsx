import React from 'react';
import { Meta, Story } from '@storybook/react';
import mockOrders from '../../mocks/mockOrders.json';
import { AddressView } from './AddressView';

export default {
  title: 'Components/AddressView',
  component: AddressView,
} as Meta;

const Template: Story = (args) => <AddressView address={args.address} />;

export const AddressViewStory = Template.bind({});
AddressViewStory.storyName = 'AddressView';
AddressViewStory.args = { address: mockOrders[0].shippingAddress };
