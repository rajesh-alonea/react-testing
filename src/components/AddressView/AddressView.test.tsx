import React from 'react';
import mockOrders from '../../mocks/mockOrders.json';
import { render } from '../../test/test-utils';
import { AddressView } from './AddressView';

describe('<AddressView />', () => {
  test('renders correctly', async () => {
    const { asFragment } = render(
      <AddressView address={mockOrders[0].shippingAddress} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
