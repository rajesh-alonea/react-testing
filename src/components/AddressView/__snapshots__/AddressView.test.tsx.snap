// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddressView /> renders correctly 1`] = `
<DocumentFragment>
  <div>
    <h3
      class="m-0"
    >
      Elon Musk
    </h3>
    <p
      class="m-0"
    >
      Tesla, Inc.
    </p>
    <p
      class="m-0"
    >
      3500 Deer Creek Road
    </p>
    <p
      class="m-0"
    >
      Palo Alto, CA 94304
    </p>
  </div>
</DocumentFragment>
`;
